server/server.go

```diff
@@ -252,6 +254,23 @@ func NewServer(logger *zap.Logger, version string, host string, port int, fronte
 	return s
 }
 
+// LoadLicense will get and parse valid license and set into application.
+func (server *Server) LoadLicense(licenseService enterprise.LicenseService) {
+	license, err := licenseService.LoadLicense()
+	if err != nil {
+		server.l.Warn("Failed to load license", zap.String("error", err.Error()))
+	} else {
+		server.l.Info(
+			"Load valid license",
+			zap.String("plan", license.Plan.String()),
+			zap.Time("expiresAt", time.Unix(license.ExpiresTs, 0)),
+			zap.Int("instanceCount", license.InstanceCount),
+		)
+	}
+
+	server.license = license
```

Comment:
We may only set license if successfully loaded. On failure, we may better to leave the current license as is instead of nil the license (I also feel this is the main reason we need to keep a license field in memory, otherwise we may just read license from file every time requesting its info)